### ISSUE TITLE
Nan value for loss

### ISSUE THREAD

**Issue by chulaihunde** [nicola-decao/s-vae-pytorch#1]
s-vae-pytorch-master/examples/mnist.py
Modify the code：

def train(model, optimizer):
    for epoch in range(100):
        for i, (x_mb, y_mb) in enumerate(train_loader):

##### Hyper-spherical VAE #####
Epoch: 0   Loss: 169.04999
Epoch: 1   Loss: 157.91183
Epoch: 2   Loss: 149.1766
Epoch: 3   Loss: nan
Epoch: 4   Loss: nan
Epoch: 5   Loss: nan
Epoch: 6   Loss: nan
Epoch: 7   Loss: nan
Epoch: 8   Loss: nan
Epoch: 9   Loss: nan
Epoch: 10   Loss: nan
Epoch: 11   Loss: nan
Epoch: 12   Loss: nan

**Comment by oskopek** [comment]
This issue for me occurs mostly with hyperspherical VAEs with Z_DIM = 2. See proposed fix: #8
